### PR TITLE
CLOUDSTACK-9168: Testpath to check if wrong value is inserted into nics table netmask field when creating a VM

### PR DESCRIPTION
**Pull request by pritisarap12**
Testcase Result:

Check Netmask value in database ... === TestName: test_01_netmask_value_check | Status : SUCCESS ===
ok

---

Ran 1 test in 143.944s

OK

### CODE REVIEW

File: test/integration/testpaths/testpath_netmask.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        return
+
+    @attr(tags=["basic"], required_hardware="false")
+    def test_01_netmask_value_check(self):
+        """ Check Netmask value in database
+            1. Check if netmask attribute in nics table
+                stores correct value.
+        """
+
+        qryresult_netmask = self.dbclient.execute(
```
Comment (sanju1010) [inline test/integration/testpaths/testpath_netmask.py]: How can you make sure that this sql query would return the netmak belongs to the vm you had created? It is possible to have more than one isolated network with same CIDR. While running this test if there was already a vm created in a network with same CIDR, then this qurey would return multiple results.

File: test/integration/testpaths/testpath_netmask.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        self.assertNotEqual(
+            len(qryresult_netmask),
+            0,
+            "Check if netmask attribute in nics table \
+            stores correct value")
+
+        netmask_id = qryresult_netmask[0][2]
+        if netmask_id != "NULL":
+            netmask_value = re.match(
```
Comment (sanju1010) [inline test/integration/testpaths/testpath_netmask.py]: This regex only checks for valid netmask. However, it does not validate whether the netmask is from the guest cidr.

**Comment by pritisarap12** [comment]
Test result after modifications:

Check Netmask value in database ... === TestName: test_01_netmask_value_check | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 121.854s

OK

**Comment by pritisarap12** [comment]
Updated the testpath with review comments:
    --Validation of netmask 
        --check it is from correct CIDR

**Comment by pavanb018** [comment]
I have reviewed the code and it looks good to me as per the code review.

**Comment by sanju1010** [comment]
-1
This test is not passing on a basic zone. In basic zone we don't list the network cidr in list network response. So test is failing at line no: 163. It only works for shared networks in advanced zone. So please add following steps in the test:
1.Create a shared network in advanced zone (Change the tagging to advanced)
2.Deploy vm in the shared network
3.Do the validation

**Comment by pritisarap12** [comment]
Changed the tags to advanced zone:

Testcase result:
Check Netmask value in database ... === TestName: test_01_netmask_value_check | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 122.485s

OK

**Comment by rhtyd** [comment]
@pritisarap12 please modify the test to be able to run on basic zone as well.

rebase against master thanks

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 76
 Hypervisor xenserver
 NetworkType Advanced
 Passed=72
 Failed=1
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vm_life_cycle.py

 * test_10_attachAndDetach_iso Failing since 2 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Pull request closed without merging**